Document Release notes for SBO 1.2 (#50453)
openshift/openshift-docs: modules/sbo-release-notes-1-2.adoc

[id="sbo-release-notes-1-2_{context}"]
// Module included in the following assembly:
//
// * applications/connecting_applications_to_services/sbo-release-notes.adoc
:_content-type: REFERENCE
= Release notes for {servicebinding-title} 1.2

{servicebinding-title} 1.2 is now available on {product-title} 4.7, 4.8, 4.9, 4.10, and 4.11.

[id="new-features-1-2_{context}"]
== New features
This section highlights what is new in {servicebinding-title} 1.2:

* Enable {servicebinding-title} to consider optional fields in the annotations by setting the `optional` flag value to `true`.
* Support for `servicebinding.io/v1beta1` resources.
* Improvements to the discoverability of bindable services by exposing the relevant binding secret without requiring a workload to be present.

[id="known-issues-1-2_{context}"]
== Known issues

> * By default, the projected files get their permissions set to 0644. {servicebinding-title} cannot set specific permissions due to a bug in Kubernetes that causes issues if the service expects specific permissions such as, `0600`. As a workaround, you can modify the code of the program or the application that is running inside a workload resource to copy the file to the `/tmp` directory and set the appropriate permissions. link:https://issues.redhat.com/browse/APPSVC-1127[APPSVC-1127]

Reviewer: 💡 RedHat.Usage 15(124-130) Verify your use of 'due to' with the word usage guidelines.
💡 RedHat.SimpleWords 15(256-262) Use simple language. Consider using 'change' rather than 'modify'.

+
.Example error message
[source,text]
----
`postgresclusters.postgres-operator.crunchydata.com "hippo" is forbidden:
        User "system:serviceaccount:my-petclinic:service-binding-operator" cannot
        get resource "postgresclusters" in API group "postgres-operator.crunchydata.com"
        in the namespace "my-petclinic"`
----
+
Workaround 1: Install the {servicebinding-title} in the `all namespaces` installation mode. As a result, the appropriate cluster-scoped RBAC rule now exists and the binding succeeds.
+
Workaround 2: If you cannot install the {servicebinding-title} in the `all namespaces` installation mode, install the following role binding into the namespace where the {servicebinding-title} is installed: 
+
.Example: Role binding for Crunchy Postgres Operator
[source,yaml]
----
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: service-binding-crunchy-postgres-viewer
subjects:
  - kind: ServiceAccount
    name: service-binding-operator
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: service-binding-crunchy-postgres-viewer-role
----
link:https://issues.redhat.com/browse/APPSVC-1062[APPSVC-1062]

> * Currently, when you modify the `ClusterWorkloadResourceMapping` resources, the {servicebinding-title} does not implement correct behavior. As a workaround, perform the following steps:

Reviewer 2: I think we need to specify what correct or incorrect is here

Author: Hi, @sadlerap: PTAL at David's comment. Please provide the required inputs. Thanks!

Reviewer: ```suggestion
* Currently, when you modify the `ClusterWorkloadResourceMapping` resources, the {servicebinding-title} does not implement the correct behavior. As a workaround, perform the following steps:
```

Reviewer 3: Maybe something like this:
```suggestion
* Currently, when you modify the `ClusterWorkloadResourceMapping` resources, the {servicebinding-title} does not implement correct behavior. When modified, the specification requires that workloads have their binding data de-projected using the old `ClusterWorkloadResourceMapping` resource, then re-projected using the new resource. As a workaround, perform the following steps:
```

Author: @sadlerap According to the specification when you change the `ClusterWorkloadResourceMapping` resources, the {servicebinding-title} must use the previous version of the `ClusterWorkloadResourceMapping` resource to remove the binding data that was getting projected till now. Currently, when you change the `ClusterWorkloadResourceMapping` resources, the {servicebinding-title} uses the latest version of the `ClusterWorkloadResourceMapping` resource to remove the binding data. As a result, {servicebinding-title} might remove the binding data incorrectly. As a workaround, perform the following steps:

Author: Done.

> * According to the specification when you change the `ClusterWorkloadResourceMapping` resources, {servicebinding-title} must use the previous version of the `ClusterWorkloadResourceMapping` resource to remove the binding data that was getting projected till now. Currently, when you change the `ClusterWorkloadResourceMapping` resources, the {servicebinding-title} uses the latest version of the `ClusterWorkloadResourceMapping` resource to remove the binding data. As a result, {the servicebinding-title} might remove the binding data incorrectly. As a workaround, perform the following steps:

Reviewer: ```suggestion
* According to the specification, when you change the `ClusterWorkloadResourceMapping` resources, {servicebinding-title} must use the previous version of the `ClusterWorkloadResourceMapping` resource to remove the binding data that was being projected until now. Currently, when you change the `ClusterWorkloadResourceMapping` resources, the {servicebinding-title} uses the latest version of the `ClusterWorkloadResourceMapping` resource to remove the binding data. As a result, {the servicebinding-title} might remove the binding data incorrectly. As a workaround, perform the following steps:
```

Author: Done

[id="sbo-release-notes-1-2_{context}"]
// Module included in the following assembly:
//
// * applications/connecting_applications_to_services/sbo-release-notes.adoc
:_content-type: REFERENCE
= Release notes for {servicebinding-title} 1.2

{servicebinding-title} 1.2 is now available on {product-title} 4.7, 4.8, 4.9, 4.10, and 4.11.

[id="new-features-1-2_{context}"]
== New features
This section highlights what is new in {servicebinding-title} 1.2:

* Enable {servicebinding-title} to consider optional fields in the annotations by setting the `optional` flag value to `true`.
* Support for `servicebinding.io/v1beta1` resources.
* Improvements to the discoverability of bindable services by exposing the relevant binding secret without requiring a workload to be present.

[id="known-issues-1-2_{context}"]
== Known issues

> * Currently, when you install {servicebinding-title} on {product-title} 4.11, the memory footprint of {servicebinding-title} increases beyond expected limits. But with low usage, the memory footprint stays within the expected ranges of your environment or scenarios. In comparison with {product-title} 4.10, under stress, both the average and maximum memory footprint increase considerably. This issue is evident in the previous versions of {servicebinding-title} as well. There is currently no workaround for this issue.

Author: @dperaza4dustbit and @pmacik: PTAL at this newly added Known Issue and provide feedback. 

Please provide the APPSVC JIRA link to this issue. I will add it here. Thanks!

Reviewer 4: Hi @Srivaralakshmi the JIRA issue is [APPSVC-1200](https://issues.redhat.com/browse/APPSVC-1200).

Author: @pmacik Thanks for the issue link. Does the RN Text look good? If yes, then pls provide your LGTM in the PR and I will add it to the JIRA.

Author: Done.

> * Currently, when you install {servicebinding-title} on {product-title} 4.11, the memory footprint of {servicebinding-title} increases beyond expected limits. But with low usage, the memory footprint stays within the expected ranges of your environment or scenarios. In comparison with {product-title} 4.10, under stress, both the average and maximum memory footprint increase considerably. This issue is evident in the previous versions of {servicebinding-title} as well. There is currently no workaround for this issue. link:https://issues.redhat.com/browse/APPSVC-1200[APPSVC-1200]

Reviewer 5: Maybe don't start the sentence with "But ", and just say...

"With low usage, the memory.."

or maybe better:

"With low usage, however, the memory.."

Author: Done.
